src/allmydata/util/encodingutil.py

```diff
@@ -434,3 +434,16 @@ def listdir_filepath(fp):
 # be NFC-normalized.
 def normalize(namex):
     return unicodedata.normalize('NFC', namex)
+
+
+def encode_all(val, encoding='utf-8'):
+    """
+    Encode text or a dict to bytes using utf-8.
+    TODO: Consider using singledispatch.
+    """
+    if isinstance(val, dict):
+        return {
+            encode_all(key): encode_all(value)
```

Comment:
My objective here was to avoid repeating the `encode("utf-8")` any number of times... to communicate that everything is getting encoded the same way. I wrote it as a recursive function so that it could potentially also be used to encode a nested dictionary if needed. I'll just inline all the encodings.

Reply:
Fixed in 239a8c0d8.